### PR TITLE
Feature/build budget app

### PR DESCRIPTION
**Pull request by topeogunleye**
## Capstone Project
In this Milestone, I 

**Day 1**
- Set up the repository and tools.
- Createed your models according to the ERD diagram. Don't forget about validations and tests.
- Implemented the authentication and authorization.

**Day 2**

- Created the page to add a category.
- Created the home page.
- Created the page to add a transaction.
- Created the transactions page.

**Day 3**

- Made sure your app is tested adequately.
- Deploy the project, and test for final details.
- Recorded a video for my project.
- Created a good README and PR description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,169 @@
...
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# 📗 Table of Contents
 
-Things you may want to cover:
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+  - [🚀 Live Demo](#live-demo)
```
Comment (MahmoudBakr23) [inline README.md:9]: - [ ] Please make sure to deploy your app and attach the valid link to it in the README file. 😄

File: README.md [hunk 2]
```diff
@@ -1,24 +1,169 @@
-# README
+<!-- TABLE OF CONTENTS -->
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+# 📗 Table of Contents
 
-Things you may want to cover:
+- [📖 About the Project](#about-project)
```
Comment (MahmoudBakr23) [inline README.md:5]: - [x] Kindly record a short video of yourself presenting what you've accomplished so far on this great project, and attach the link in the README and PR summary. 🙏

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (MahmoudBakr23) [inline spec/models/category_spec.rb]: - [x] Please resolve any pending tests. Overall, kindly add more unit tests for your models, controller's methods, private methods, and some integration tests as well haven't been yet written! 👍

File: app/models/ability.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, published: true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
+  end
```
Comment (MahmoudBakr23) [inline app/models/ability.rb:33]: - [x] I'm getting this syntax error when I try to create a new transaction. Please fix that. 🙏 

Screenshot (1)

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @topeogunleye 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done working with the needed gems for the app such as devise and cancan gems
- Great work with the controllers' restful methods
- Well done setting up your migration files, and configuring the database with PostgreSQL
- Well done setting up the routes
- Great design that follows the given one
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 4 above)
Comment (Munsa1) [reply app/models/ability.rb:33]: - [x] As requested by the previous reviwer kindly fix this issue as I am also unable to create a new transaction
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @topeogunleye, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by m05e5 — APPROVED** [review]
**Hi @topeogunleye 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Nice UI
✔️ No Linters error

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**